src/config.js

```diff
@@ -0,0 +1,49 @@
+
+import { logWarn, logError } from 'utils';
+
+const CONFIG_ATTR = 'data-prebid-config';
+
+var scripts = document.querySelectorAll(`script[${CONFIG_ATTR}]`);
+var scriptConfig = {};
+var config = {};
+var configHandlers = {};
+
+for (let i = 0; i < scripts.length; i++) {
+  try {
+    let script = scripts[i];
+    let attrConfig = script.getAttribute(CONFIG_ATTR);
+
+    if (attrConfig) {
+      Object.assign(scriptConfig, JSON.parse(attrConfig));
+    }
+  } catch (e) {
+    logError(`Error parsing ${CONFIG_ATTR}`, null, e);
+  }
+}
+
+export function setConfig(setConfig) {
+  Object.assign(config, setConfig);
+
+  Object.keys(setConfig).forEach(property => {
+    if (Array.isArray(configHandlers[property])) {
+      configHandlers[property].forEach(handler => {
+        if (!config[property]) {
+          logWarn(`no config found for ${property}`);
+        }
+
+        handler(config[property]);
+      })
+    }
+  });
+}
+
+export function getConfig(property, handler) {
+  if (!configHandlers[property]) {
+    configHandlers[property] = [];
+  }
+  configHandlers[property].push(handler);
+}
```

Comment:
What happens if `getConfig` is called after the last call to `setConfig`? If I were calling these APIs, I would still expect my callback to be executed.

Actually... a related problem: I would expect my callback to be executed exactly once. It doesn't affect your use-cases here... but looking at this file in isolation, I'd still consider it a bug.

Related: are unit tests coming soon?